### ISSUE TITLE
consider implementing shared_driver (used in automatic driver management) as ThreadLocal (like in Selenide)

### ISSUE THREAD

**Issue by yashaka** [yashaka/selene#34]
Do we actually need this?

Does anybody run tests in parallel threads? Does it have any benefits taking into account python's GIL limitation?

**Comment by SergeyPirogov** [comment]
Have checked selene and parallel test execution using pytest-xdist - everything work perfect independent on number of threads

**Comment by yashaka** [comment]
Hm... How does it work then? Maybe it work just because pytest-xdist does
multi-processing not multithreading, doesn't it?

**Comment by SergeyPirogov** [comment]
Yes, exactly. I am not sure that people use another approach in python world

**Comment by yashaka** [comment]
Ok, great, then we do not need anything to do here so far, and can close
this issue!

**Comment by zhupengfarmer** [comment]
@yashaka use pytest-parallel  ui auto  test    is   thread rather than process, so selene does not support thread concurrency for automation execution

**Comment by yashaka** [comment]
@zhupengfarmer, Let's continue discussion in new ticket  #475 that you have created!
